ros2_generate_interface_docs/ros2_generate_interface_docs/main.py

```diff
@@ -14,11 +14,82 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import argparse
+import os
 import sys
+import time
+
+from ros2_generate_interface_docs import msg_utils
+from ros2_generate_interface_docs import utils
+
+from rosidl_runtime_py import get_message_interfaces
+
+
+def generate_interfaces(interfaces, html_dir, template, interface_type):
+    """
+    Generate the index and documentation for each message.
+
+    Parameters
+    ----------
+    interfaces: dict of {str : str[]}
+        dictionary with the interface package name associated with all the interface
+        for this package.
+    html_dir: str
+        path to the directory to save the generated documentation
+    template: str
+        template name
+    interface_type: str
+        type of the interface: msg, action or srv
+
+    """
+    for package_name, interface_names in sorted(interfaces.items(), key=lambda item: item[0]):
+        package_directory = os.path.join(html_dir, package_name)
+        interface_type_directory = os.path.join(package_directory, interface_type)
+        os.makedirs(interface_type_directory, exist_ok=True)
+        utils.generate_index(package_name,
+                             package_directory,
+                             interfaces[package_name])
+        for interface_name in interface_names:
+            documentation_data = {'name': interface_name,
+                                  'package': package_name,
+                                  'base_type': interface_name,
+                                  'date': str(time.strftime('%a, %d %b %Y %H:%M:%S'))}
```

Comment:
It would be good to use the same timestamp for all files and not generate a new one for each file.